src/MTProtoTools/FilesLogic.php

```diff
@@ -328,12 +328,10 @@ public function uploadFromStream(mixed $stream, int $size, string $mime, string
                 $created = true;
             }
             $callable = static function (int $offset, int $size) use ($stream) {
-                $reader = $stream->getReadBuffer($l);
                 try {
-                    return $reader->bufferRead($size);
+                    return $stream->read(null, $size);
```

Comment:
Did not tested upload. Will do later

Reply:
BTW, this should still be bufferRead, because $stream is a BufferedRawStream